lib/rucio/common/stomp_utils.py

```diff
@@ -15,145 +15,415 @@
 
-    def is_stalled(self, connection: Connection, *, logger: "LoggerFunction" = logging.log):
-        if not connection.is_connected():
-            return True
+        Args:
+            config_section (str): The name of the config section for this manager to parse for configuration.
+            logger (logging.Logger): logger to use. Defaults to logging.getLogger(__name__).getChild(__qualname__).
+        """
+        if logger is not None:
+            self._logger = logger
+        self._config = self._parse_config(config_section)
+        self._listener_factory = None
+        self._conns = []
+        for broker in self._config.brokers:
+            conn = Connection(host_and_ports=[broker],
+                              vhost=self._config.vhost,
+                              reconnect_attempts_max=self._config.reconnect_attempts_max,
+                              timeout=self._config.timeout,
+                              heartbeats=self._config.heartbeats)
+            if self._config.use_ssl:
+                conn.set_ssl(cert_file=self._config.cert_file, key_file=self._config.key_file)
+            self._conns.append(conn)
 
-        if self._heartbeats and getattr(connection, 'received_heartbeat') and connection.received_heartbeat:
-            heartbeat_period_seconds = max(0, self._heartbeats[0], self._heartbeats[1]) / 1000
+    @property
+    def config(self) -> StompConfig:
+        """
+        Get the config.
 
-            if not heartbeat_period_seconds:
-                return False
+        Returns:
+            StompConfig: config object.
+        """
+        return deepcopy(self._config)
 
-            now = monotonic()
-            if connection.received_heartbeat + 10 * heartbeat_period_seconds < now:
-                logger(logging.WARNING, "Stomp connection missed heartbeats for a long time")
-                return True
+    def set_listener_factory(self, name: str, listener_cls: type, **kwargs):
+        """
+        Setup listener factory
 
-        return False
+        This method will setup a factory to create a name and listener for the arguments to
+        connection.set_listener based on pre-defined argument values.
 
-    def disconnect(self):
-        for conn in self._connections.values():
+        Args:
+            name (str): Listener name
+            listener_cls (ListenerBase): Listener class.
+        """
+        def create_listener(name, listener_factory, conn):
+            return name, listener_factory(conn=conn)
+        self._listener_factory = partial(create_listener,
+                                         name=name,
+                                         listener_factory=partial(listener_cls, logger=self._logger, **kwargs))
+
+    def _parse_config(self, config_section: str) -> StompConfig:
+        """
+        Parse config section.
+
+        Args:
+            config_section (str): The name of the config section for this manager to parse for configuration.
```

Comment:
Please avoid specifying the argument types in the docstring if they are already present as type annotations; there's no need to duplicate this information and it can cause mismatches between the type annotation and the type specified in the docstring if changes are made down the line